packages/dashboard/src/app/views/myStories/content/storyGridView/index.js

```diff
@@ -136,31 +136,42 @@ const StoryGridView = ({
 
   useFocusOut(containerRef, () => setActiveGridItemId(null), []);
 
+  const getModifiedMenuItem = useCallback(
+    (value) => {
+      const { action, ...rest } = storyMenu.menuItems.find(
+        (item) => item?.value === value
+      );
+
+      return {
+        action: (story) => {
+          manuallySetFocusOut();
+          action(story);
+        },
+        ...rest,
+      };
+    },
+    [manuallySetFocusOut, storyMenu.menuItems]
+  );
+
   const modifiedStoryMenu = useMemo(() => {
+    const actions = [
+      STORY_CONTEXT_MENU_ACTIONS.DELETE,
+      STORY_CONTEXT_MENU_ACTIONS.RENAME,
+      STORY_CONTEXT_MENU_ACTIONS.DUPLICATE,
+    ];
+
+    const menuItems = storyMenu.menuItems.map((item) => {
+      return actions.includes(item?.value)
+        ? getModifiedMenuItem(item?.value)
+        : item;
+    });
```

Comment:
```suggestion
    const menuItems = storyMenu.menuItems.map((item) => {
      if (actions.includes(item?.value)) {
        item.action = (story) => {
          manuallySetFocusOut();
          item.action(story);
        };
      }

      return item;
    });
```

If we do this then we don't need `getModifiedMenuItem`

Reply:
Maybe it's creating some infinite loop somewhere 🤔 , checking...

Reply:
When using a keyboard shortcut to delete the story

![image](https://user-images.githubusercontent.com/6297436/150147440-cc8a4927-b8aa-488a-8bec-f594711cec73.png)

Reply:
In fact, even when clicking the delete button it creates an infinite loop.

Reply:
Fixed now, waiting for tests to pass.